src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -603,11 +602,13 @@ object DeepDiveLogCompiler extends DeepDiveLogHandler {
       var weight = ""
       for (cqBody <- stmt.q.bodies) {
         // edge query
-        val fakeHeadAsBody  = BodyAtom(stmt.q.head.name, stmt.q.head.terms map {
-          case x: VarExpr => VarPattern(x.name)
-          case x: Expr    => ExprPattern(x)
-        })
-        val fakeBody        = fakeHeadAsBody +: cqBody
+        val headAsBody = stmt.head.terms map { x =>
+          BodyAtom(x.name, x.terms map {
+            case x: VarExpr => VarPattern(x.name)
+            case x: Expr    => ExprPattern(x)
+          })
+        }
+        val fakeBody        = headAsBody ++ cqBody
```

Comment:
This looks like some clever hack, which may be okay for the short term.  However, a comment explaining why we turn a head into a body would be extremely helpful for someone who wants to do the right thing later.